src/astro/sql/parsers/sql_directory_parser.py

```diff
@@ -6,22 +7,91 @@
 from airflow.decorators.base import get_unique_task_id
 from airflow.decorators.task_group import task_group
 from airflow.exceptions import AirflowException
+from airflow.models.dag import DagContext
 from airflow.models.xcom_arg import XComArg
 
 from astro.sql.operators.sql_decorator import SqlDecoratedOperator
 from astro.sql.table import Table, TempTable
 
 
+def get_path_for_render(path):
+    """
+    Generate a path using either the DAG's template_searchpath with a relative path, or with the relative path
+    to the DAG file
+
+    :param path:
+    :return:
+    """
+    template_path = DagContext.get_current_dag().template_searchpath
+    if template_path:
+        for t in template_path:
+            try:
+                os.listdir(t + "/" + path)
+                logging.info("Template_path found. rendering %s", t + "/" + path)
+                return t + "/" + path
+            except FileNotFoundError:
+                logging.info("Could not find template_path %s", t + "/" + path)
+    logging.info("No template path found, rendering base path %s", path)
+    return path
+
+
 def render(
-    path,
+    path: str = "",
     conn_id: Optional[str] = None,
     database: Optional[str] = None,
     schema: Optional[str] = None,
     warehouse: Optional[str] = None,
     role: Optional[str] = None,
     **kwargs,
 ):
-    # raise AirflowException(f"Failed because cwd is {os.listdir(path)}, {os.}")
+    """
+    aql.render renders a single sql file or a directory of sql files and turns them into a group of tasks with
+    dependencies. The result of this function is a dictionary of tasks that you can reference in downstream tasks.
+
+    e.g.:
+
+    dag = DAG(
+        dag_id="example_postgres_render",
+        start_date=datetime(2019, 1, 1),
+        max_active_runs=3,
+        schedule_interval=timedelta(minutes=30),
+        default_args=default_args,
+        template_searchpath=dir_path,
+    )
+
+
+    @adf
+    def print_results(df: pd.DataFrame):
+        print(df.to_string())
```

Comment:
This forces the user who is learning to use `render` to also have to learn about the Dataframe operator. Would there be a simpler way of showing the benefit of `render`?

Reply:
@tatiana I'd imagine the only two functions that would make sense here are transform and dataframe. No  preference on which. The dataframe function IMO is cooler because it shows you can move the SQL into python